packages/react-components/react-tags/src/contexts/tagGroupContext.tsx

```diff
@@ -12,7 +12,14 @@ const tagGroupContextDefaultValue: TagGroupContextValue = {
  * Context shared between TagGroup and its children components
  */
 export type TagGroupContextValue = Required<Pick<TagGroupState, 'handleTagDismiss' | 'size'>> &
-  Partial<Pick<TagGroupState, 'appearance' | 'dismissible'>>;
+  Partial<Pick<TagGroupState, 'appearance' | 'dismissible'>> & {
+    /**
+     * Boolean that indicates that the tag is being render inside of a
+     * TagPicker element.
+     * When rendered inside of a TagPicker a tag should have role="option"
+     */
+    isInsideTagPicker?: boolean;
```

Comment:
It might be better to handle the top level toolbar/listbox role on the TagGroup and pass that down through ctx to the Tags which set the appropriate role based on parent. It would also play nicer with the fact that the TagPicker currently recomposes TagGroup to set that role

Reply:
Sounds like a good idea! I'll implement that instead